assets/src/components/Sidebar/CampaignSidebar.js

```diff
@@ -74,16 +75,34 @@ export class CampaignSidebar extends Component {
     super( props );
     this.state = {
       theme: null,
+      meta: null,
     };
     this.handleThemeChange = this.handleThemeChange.bind( this );
   }
 
   componentDidMount() {
-    const meta = wp.data.select( 'core/editor' ).getEditedPostAttribute( 'meta' );
-    if (meta) {
-      const theme = meta[ 'theme' ];
-      this.loadTheme( theme );
-    }
+    wp.data.subscribe( () => {
+      const meta = wp.data.select( 'core/editor' ).getEditedPostAttribute( 'meta' );
+      if (meta) {
+        let theme = meta[ 'theme' ];
+        if ( theme === '' ) {
+          theme = 'default';
+        }
+        if (!isShallowEqual(this.state.meta, meta)) {
+          this.setState( prevState => {
+            return {
+              meta: meta,
+              theme: prevState.theme,
```

Comment:
Normally I would have used `...prevState` here instead of repeating `theme`, however this caused an error which I think is a bug with the `...` operator.